modules/core/src/main/java/org/apache/ignite/internal/util/VersatileReadWriteLock.java

```diff
@@ -364,16 +364,31 @@ private void notifyWriteLockSolicitors() {
         for (Iterator<CompletableFuture<Void>> iterator = writeLockSolicitors.iterator(); iterator.hasNext(); ) {
             CompletableFuture<Void> future = iterator.next();
 
-            if (!tryWriteLock()) {
+            if (tryWriteLock()) {
```

Comment:
Shall we call `iterator.next` after taking the lock?

Reply:
No, why?

Reply:
Futures represent acqusition intents. We have just acquired the write lock for this intent, so we should remove it from the waiting set (as its acqusition request has been satisfied)

Reply:
Because the scope is too wide, you only need the future after you successfully obtained the lock, no?

Reply:
Your sentence consists of two parts (about scope and about the future). I did not understand either of them :) Could you please elaborate?

Reply:
I see, it's about changing the order: first taking the lock, then invoking `next()`. I implemented this change